src/sagemaker/config/config_schema.py

```diff
@@ -477,6 +491,33 @@ def _simple_path(*args: str):
                             TYPE: OBJECT,
                             ADDITIONAL_PROPERTIES: False,
                             PROPERTIES: {
+                                SESSION: {
+                                    TYPE: OBJECT,
+                                    ADDITIONAL_PROPERTIES: False,
+                                    PROPERTIES: {
+                                        S3_BUCKET: {
+                                            "description": "sets `default_bucket` of Session",
+                                            "$ref": "#/definitions/s3Bucket",
+                                        },
+                                        S3_OBJECT_KEY_PREFIX: {
+                                            "description": (
+                                                "sets `default_bucket_prefix` of Session"
+                                            ),
+                                            TYPE: "string",
+                                            # Regex based on
+                                            # https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+                                            # For now, the regex only allows the "safe characters"
+                                            # specified by S3. The regex can be loosened (but not
+                                            # tightened) in the future without introducing
+                                            # backward incompatibility.
+                                            "pattern": (
+                                                r"^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$"
```

Comment:
I couldn't find this pattern in the public docs, lets check up on this offline

Reply:
removed regex and handled the main edge cases I was worried about by refactoring the S3 utils